torch/distributed/algorithms/ddp_comm_hooks/default_hooks.py

```diff
@@ -65,3 +65,31 @@ def decompress(fut):
         return [decompressed_tensor]
 
     return fut.then(decompress)
+
+def fp16_compress_wrapper(hook: callable) -> callable:
+    """
+    This wrapper applies ``float 32`` to ``float 16`` compression to an arbitrary DDP communication hook.
```

Comment:
Don't mention FP32 here. The input data type can be FP64 or other types, as long as it can be casted into FP16. I will submit another PR to update the docstring of `fp16_compress_hook` as well.

I would reword it like this:
"This wrapper casts the input gradient tensors of a given DDP communication hook to half-precision floating point format (``torch.float16``), and casts the resulting tensors of the given hook back to the input data type such as ``float32``.